### PR TITLE
Check if auth config is provided when using access control

### PR DESCRIPTION
**Pull request by peusebiu**

### CODE REVIEW

File: pkg/api/authz.go [hunk 1]
```diff
@@ -228,6 +228,16 @@ func isBearerAuthEnabled(config *Config) bool {
 	return false
 }
 
+func isAuthorizationEnabled(config *Config) bool {
```
Comment (rchincha) [inline pkg/api/authz.go:231]: Call this isAuthnEnabled()

Comment (peusebiu) [reply pkg/api/authz.go]: Fixed, thank you!

File: pkg/cli/root.go [hunk 2]
```diff
@@ -164,6 +164,14 @@ func LoadConfiguration(config *api.Config, configPath string) {
 		panic(errors.ErrBadConfig)
 	}
 
+	// check authorization config, it should have basic auth enabled or ldap
+	if config.HTTP.RawAccessControl != nil {
+		if config.HTTP.Auth == nil || (config.HTTP.Auth.HTPasswd.Path == "" && config.HTTP.Auth.LDAP == nil) {
+			log.Error().Err(errors.ErrBadConfig).Msg("Bad configuration, authorization config without BASIC auth or LDAP")
```
Comment (rchincha) [inline pkg/cli/root.go:170]: Msg("access control config requires htpassd or ldap authentication to be enabled")

File: pkg/api/authz.go [hunk 3]
```diff
@@ -228,6 +228,16 @@ func isBearerAuthEnabled(config *Config) bool {
 	return false
 }
 
+func isAuthnEnabled(config *Config) bool {
+	if config.AccessControl != nil &&
```
Comment (rchincha) [inline pkg/api/authz.go:232]: Remove this access control "if"

File: pkg/api/routes.go [hunk 4]
```diff
@@ -55,7 +55,7 @@ func NewRouteHandler(c *Controller) *RouteHandler {
 func (rh *RouteHandler) SetupRoutes() {
 	rh.c.Router.Use(AuthHandler(rh.c))
 
-	if !isBearerAuthEnabled(rh.c.Config) && rh.c.Config.AccessControl != nil {
+	if !isBearerAuthEnabled(rh.c.Config) && isAuthnEnabled(rh.c.Config) {
```
Comment (rchincha) [inline pkg/api/routes.go:58]: After fixing isAuthnEnabled() above, 

if rh.c.Config.AccessControl != nil && isAuthnEnabled() && !isBearerAuthEnabled {
}

Comment (rchincha) [reply pkg/api/routes.go:58]: Also add a log here saying access control is being enabled

File: pkg/api/authz.go (hunk 3 above)
Comment (peusebiu) [reply pkg/api/authz.go:232]: Ok, I didn't understand in the first place. I also moved isAuthnEnabled and isBearerAuthEnabled in auth.go.

File: pkg/api/routes.go (hunk 4 above)
Comment (peusebiu) [reply pkg/api/routes.go]: done, thank you!

File: pkg/api/routes.go [hunk 5]
```diff
@@ -55,7 +55,8 @@ func NewRouteHandler(c *Controller) *RouteHandler {
 func (rh *RouteHandler) SetupRoutes() {
 	rh.c.Router.Use(AuthHandler(rh.c))
```
Comment (rchincha) [inline pkg/api/routes.go:57]: Add a comment here saying that "authz is being enabled because authn is found"

Comment (rchincha) [reply pkg/api/routes.go:57]: Also make sure "zot verify" catches this situation.

Comment (peusebiu) [reply pkg/api/routes.go]: I already added 2 test cases for this: https://github.com/anuvu/zot/pull/221/files#diff-c3e95b9d0ad189ba68bf30694e60fd679eb406ac541ff635f3171f18a6d794a4R88

Or do you mean another situation?

**Pull request merged**